components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/deployment/application/deployer/CappDeployer.java

```diff
@@ -0,0 +1,125 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.initializer.deployment.application.deployer;
+
+import java.io.File;
+
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.axis2.deployment.AbstractDeployer;
+import org.apache.axis2.deployment.DeploymentException;
+import org.apache.axis2.deployment.repository.util.DeploymentFileData;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.micro.application.deployer.AppDeployerUtils;
+import org.wso2.micro.application.deployer.CarbonApplication;
+import org.wso2.micro.core.util.FileManipulator;
+
+public class CappDeployer extends AbstractDeployer {
+
+    private static final Log log = LogFactory.getLog(CappDeployer.class);
+
+    private AxisConfiguration axisConfig;
+
+    /**
+     * Carbon application repository directory
+     */
+    private String cAppDir;
+
+    /**
+     * Carbon application file directory (i.e. '.car')
+     */
+    private String extension;
+
+    public void init(ConfigurationContext configurationContext) {
```

Comment:
Where are we calling this?

Reply:
inside addCAppDeployer() in AppDeployerServiceComponent class.